File: 12-integer-to-roman/12-integer-to-roman.cpp
class Solution {
public:
    string intToRoman(int num) {
        string res = "";
        
        num = build_res(res, num, 1000, "M");
        num = build_res(res, num, 900, "CM");
        num = build_res(res, num, 500, "D");
        num = build_res(res, num, 400, "CD");
        num = build_res(res, num, 100, "C");
        num = build_res(res, num, 90, "XC");
        num = build_res(res, num, 50, "L");
        num = build_res(res, num, 40, "XL");
        num = build_res(res, num, 10, "X");
        num = build_res(res, num, 9, "IX");
        num = build_res(res, num, 5, "V");
        num = build_res(res, num, 4, "IV");
        num = build_res(res, num, 1, "I");
        
        
        return res;
    }
    
    int build_res(string& res, int num, int divident, string sym){
        int temp = num / divident;
        for(int i = 0; i < temp; i++)
            res = res + sym;
        
        return num % divident;
    }
};

File: 12-integer-to-roman/12-integer-to-roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        result = ''
        
        result += 'M' * (num // 1000)
        num %= 1000

        result += 'CM' * (num // 900)
        num %= 900

        result += 'D' * (num // 500)
        num %= 500

        result += 'CD' * (num // 400)
        num %= 400

        result += 'C' * (num // 100)
        num %= 100

        result += 'XC' * (num // 90)
        num %= 90

        result += 'L' * (num // 50)
        num %= 50

        result += 'XL' * (num // 40)
        num %= 40

        result += 'X' * (num // 10)
        num %= 10

        result += 'IX' * (num // 9)
        num %= 9

        result += 'V' * (num // 5)
        num %= 5

        result += 'IV' * (num // 4)
        num %= 4

        result += 'I' * num

        return result 